ckanext/cloudstorage/fanstatic/scripts/cloudstorage-multipart-upload.js

```diff
@@ -241,13 +242,24 @@ ckan.module('cloudstorage-multipart-upload', function($, _) {
                 return;
             }
             event.preventDefault();
-
-            try{
-                this._onDisableSave(true);
-                this._onSaveForm();
-            } catch(error){
-                console.log(error);
+            var dataset_id = this.options.packageId;
+            this._btnClick = $(event.target).attr('value');
+            if (this._btnClick == 'go-dataset') {
                 this._onDisableSave(false);
+                var redirect_url = this.sandbox.url(
+                    '/dataset/edit/' +
+                    dataset_id);
+                window.setTimeout(function(){
+                    window.location = redirect_url;
+                }, 1000);
+            } else {
```

Comment:
Wouldn't it be better, if you just return from this function in case of `go-dataset` click, instead of nesting different cases?